src/cryptonote_core/pulse.cpp

```diff
@@ -932,7 +942,10 @@ round_state prepare_for_round(round_context &context, service_nodes::service_nod
     auto now                     = pulse::clock::now();
     auto const time_since_block  = now <= context.wait_for_next_block.round_0_start_time ? std::chrono::seconds(0) : (now - context.wait_for_next_block.round_0_start_time);
     size_t round_usize           = time_since_block / service_nodes::PULSE_ROUND_TIME;
-    uint8_t curr_round           = static_cast<uint8_t>(round_usize); // TODO: Overflow check
+    uint8_t curr_round           = static_cast<uint8_t>(round_usize);
+
+    if (round_usize > 255) // Network stalled
+      return round_state::wait_for_next_block;
```

Comment:
It doesn't really affect anything, but it feels like `curr_round` should be set from `round_usize` after the `round_usize` value check.

Reply:
Done